hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaSelfDestructOperation.java

```diff
@@ -72,6 +72,12 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
 		if (updater.contractIsTokenTreasury(address)) {
 			return reversionWith(updater.get(beneficiaryAddress), HederaExceptionalHaltReason.CONTRACT_IS_TREASURY);
 		}
+		if (updater.contractHasAnyBalance(address)) {
+			return reversionWith(updater.get(beneficiaryAddress), HederaExceptionalHaltReason.TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);
+		}
+		if (updater.contractOwnsNfts(address)) {
+			return reversionWith(updater.get(beneficiaryAddress), HederaExceptionalHaltReason.CONTRACT_STILL_OWNS_NFTS);
+		}
 		if (address.equals(beneficiaryAddress)) {
 			return reversionWith(updater.get(beneficiaryAddress), HederaExceptionalHaltReason.SELF_DESTRUCT_TO_SELF);
 		} else {
```

Comment:
Can we extract a `validateFalse()`-like method for all of these `if`s here, looks like their structure may allow it and would make the code cleaner?

Reply:
If we were to throw an exception when any of these checks fails, a validateFalse()-like method would be suitable. In this case we would just end up wrapping an error result and then checking whether it is present or not, which would cause a bit more clutter than what we have now.